frontends/web/src/utils/androidtransport.ts

```diff
@@ -16,40 +15,39 @@ import { runningInAndroid } from './env';
  * limitations under the License.
  */
 
-let queryID = 0;
-let queryPromises = {};
-let currentListeners = [];
+import { TMsgCallback, TQueryPromiseMap } from './transport-common';
+import { runningInAndroid } from './env';
 
-/* The following line declares this as a global variable to eslint */
-/* global android */
+let queryID: number = 0;
+const queryPromises: TQueryPromiseMap = {};
+const currentListeners: Function[] = [];
 
-export function androidCall(query) {
+export function androidCall(query: string) {
   return new Promise((resolve, reject) => {
     if (runningInAndroid()) {
-      // @ts-ignore
       if (typeof window.onAndroidCallResponse === 'undefined') {
-        // @ts-ignore
-        window.onAndroidCallResponse = (queryID, response) => {
+        window.onAndroidCallResponse = (
+          queryID: number,
+          response: string,
+        ) => {
           queryPromises[queryID].resolve(response);
           delete queryPromises[queryID];
         };
       }
-
       queryID++;
       queryPromises[queryID] = { resolve, reject };
-      // @ts-ignore
-      android.call(queryID, query);
+      if (window.android) {
+        window.android.call(queryID, query);
+      }
```

Comment:
window.android!.call(queryID, query) - we know that window.android exists at this point